httplogs/httplog.go

```diff
@@ -0,0 +1,110 @@
+package httplogs
+
+import (
+	"bytes"
+	"fmt"
+	"net/http"
+	// "strconv"
+	"time"
+
+	objConsulAgent "github.com/carwale/golibraries/consulagent"
+	"github.com/carwale/golibraries/gologger"
+)
+
+var (
+	globalConsulAgent      *objConsulAgent.ConsulAgent
+	isMonitoringLogEnabled bool
+	serviceLogger          *gologger.CustomLogger
+	globalserviceName      string
+)
+
+type BhriguResponseHeader struct {
+	superHandler http.Handler
+}
+
+// InitLoggingWrapper acts as a constructor to initialize the logging service and
+// initailize the struct
+func InitLoggingWrapper(handlerToWrap http.Handler, key string, consulIP string, logger *gologger.CustomLogger, serviceName string) *BhriguResponseHeader {
+	// fmt.Println("httplog.Constructor called")
+	setBasicConfig(key, consulIP, logger, serviceName)
+	return &BhriguResponseHeader{handlerToWrap}
+}
+
+//ServeHTTP acts as middleware and can be used to do pre/post processing
+func (rh *BhriguResponseHeader) ServeHTTP(w http.ResponseWriter, r *http.Request) {
+	// fmt.Println("httplog.ServeHTTP called")
+
+	//call the wrapped handler
+	rh.superHandler.ServeHTTP(w, r)
+}
+
+// SetBasicConfig start point of the request
+func setBasicConfig(key string, consulIP string, logger *gologger.CustomLogger, serviceName string) {
+	// fmt.Println("httplog.setBasicConfig called")
+	globalConsulAgent = objConsulAgent.NewConsulAgent(
+		objConsulAgent.ConsulHost(consulIP),
+		objConsulAgent.Logger(logger),
+	)
+	serviceLogger = logger
+	globalserviceName = serviceName
+
+	go checkHTTPLogStatus(key)
+}
+
+func checkHTTPLogStatus(key string) {
+	for {
+		// fmt.Println("**************isMonitoringLogEnabled:" + strconv.FormatBool(isMonitoringLogEnabled))
+		time.Sleep(10 * time.Second)
+
+		// Monitoring key considered here
+		monitoringLoggerTime := getValueFromConsulByKey(key)
+		loggerTime, err := time.Parse("01/02/2006 15:04:05", monitoringLoggerTime)
+
+		if err != nil {
+			isMonitoringLogEnabled = false
+		}
+
+		if loggerTime.Before(time.Now()) {
+			isMonitoringLogEnabled = false
+		}
+
+		isMonitoringLogEnabled = true
+	}
+}
+
+// LogHTTPLogs display the log based on isMonitoringLogEnabled flag
+func LogHTTPLogs(r *http.Request, statusCode int) {
```

Comment:
This function has to be called from ServeHTTP method above. The called need not know about this method